### PR TITLE
sourcegraph: Create empty index for empty repositories

### PR DESCRIPTION
**Pull request by keegancsmith**
Currently an empty repository will never show up in the list of indexed
repositories, since there is no content to index. This leads to sourcegraph
believing the repos have not been indexed which can report confusing behaviour
to users. We can index empty repositories, we just need to create empty
indexes.

### CODE REVIEW

File: cmd/zoekt-sourcegraph-indexserver/main.go [hunk 1]
```diff
@@ -84,6 +91,19 @@ func refresh(root *url.URL, indexDir string, interval time.Duration, cpuFraction
 	}
 }
 
+func createEmptyShard(indexDir, name string) {
+	cmd := exec.Command("zoekt-archive-index",
+		"-index", indexDir,
+		"-incremental",
+		"-branch", "HEAD",
+		"-commit", "404aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
```
Comment (nicksnyder) [inline cmd/zoekt-sourcegraph-indexserver/main.go:99]: what is the significance here? is this the actual commit hash of an empty archive?

**Review by nicksnyder — COMMENTED** [review]
lgtm

File: cmd/zoekt-sourcegraph-indexserver/main.go (hunk 1 above)
Comment (keegancsmith) [reply cmd/zoekt-sourcegraph-indexserver/main.go:99]: nope, its a made up commit. but it starts with 404 to be a clue.

Comment (keegancsmith) [reply cmd/zoekt-sourcegraph-indexserver/main.go:99]: I didn't check, but we may be able to not include the commit in a shard file. I wasn't sure on the implication so did this instead. On more thought I believe it should just work.

Comment (nicksnyder) [reply cmd/zoekt-sourcegraph-indexserver/main.go:99]: If you leave it in, just add a comment that it is a dummy commit id

Comment (keegancsmith) [reply cmd/zoekt-sourcegraph-indexserver/main.go:99]: updated in master

**Pull request merged**